### PR TITLE
x11-misc/qt5ct: Qt5 Configuration Tool

### PR DESCRIPTION
**Pull request by mndar**
This program allows users to configure Qt5 settings (theme, font,
icons, etc.) under DE/WM without Qt integration.

Gentoo-Bug: https://bugs.gentoo.org/577840
Reported-By: Michael Palimaka <kensington@gentoo.org>
Ebuild-By: Mandar Joshi <emailmandar@gmail.com>

### CODE REVIEW

**Comment by monsieurp** [comment]
@kensington @gentoo/kde

**Comment by idella** [comment]
could use ${PN} in SRC_URI.  You might want to call einstalldocs or dodoc ... in src_install.
It might be nice to have some pkg_postinst with how to make the package actually do something

**Comment by mndar** [comment]
Will implement all of the above. I'll run it by you first on IRC before committing to the repo.
I have my term exams this week. Once done, will contact you next week

**Comment by Pesa** [comment]
@gentoo/qt

File: x11-misc/qt5ct/qt5ct-0.22.ebuild [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.bz2"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+RDEPEND=">=dev-qt/qtcore-5.4.0
+	>=dev-qt/qtgui-5.4.0
+	>=dev-qt/qtwidgets-5.4.0"
+DEPEND="${RDEPEND}
+	>=dev-qt/linguist-tools-5.4.0"
```
Comment (rindeal) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: QT deps could be handled by `add_qt_dep()` from `kde5-functions.eclass` and setting `QT_MINIMAL=5.4`

Comment (kensington) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: I recommend against using kde eclasses for non-KDE stuff.

Comment (rindeal) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: whose fault is it that the function is not in some qt eclass or elsewhere? The scope of it is in no way limited just to kde5.

If you're worried about pre-setting variables like `HOMEPAGE` then please note that `kde5-functions.eclass` contains only functions.

File: x11-misc/qt5ct/qt5ct-0.22.ebuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+inherit qmake-utils
+
+DESCRIPTION="This program allows users to configure Qt5 settings"
+HOMEPAGE="http://qt5ct.sourceforge.net/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.bz2"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+RDEPEND=">=dev-qt/qtcore-5.4.0
```
Comment (Pesa) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: missing SLOT `:5`

Feel free to drop the `>=5.4.0` requirement, it's satisfied by every qt version in the tree.

File: x11-misc/qt5ct/qt5ct-0.22.ebuild [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	eqmake5 ${PN}.pro
+}
+
+src_install() {
+	emake INSTALL_ROOT="${D}" install
+	einstalldocs
+}
+
+pkg_postinst() {
+	elog "qt5ct allows users to configure Qt5 settings"
```
Comment (Pesa) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: I would drop this line

File: x11-misc/qt5ct/qt5ct-0.22.ebuild (hunk 1 above)
Comment (Pesa) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: this is *not* a valid reason to `inherit kde5-functions`

Comment (rindeal) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: And what would be? The eclass doesn't contain much more than the function.

Comment (Pesa) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: The eclass is kde5-specific, its maintainers don't care about qt-only use cases, and they could make assumptions that are not valid for qt-only packages, or introduce deps or change behavior in unexpected ways...

In any case, I don't see the advantage of using the eclass in the first place, so this whole discussion is pointless.

Comment (rindeal) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: no, some points have certainly been made here, eg. you've just made one or two

Comment (Pesa) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: We can certainly add a `qt5_add_dep()` to `qmake-utils.eclass` if there's enough demand for it, but that doesn't exonerate you from knowing that you must include a SLOT in your dependency atoms.

**Comment by monsieurp** [comment]
ping @mndar

**Comment by mndar** [comment]
As discussed with monsieurp and suggested by idella4 and wraeth, will consult with devs on IRC (gentoo-kde and gentoo-qt)

**Comment by mndar** [comment]
I'm keen on getting this PR fixed but as this is my final year in college and I'm too busy with my exams  the whole of this May. I'll be free only by the end of this month i.e. 30th May. I'll be able to devote my time to this immediately after that.

**Comment by mndar** [comment]
ping @Pesa 
Which IRC channel can I find you in?

**Comment by Pesa** [comment]
I'm seldom on IRC, but @kensington can also help, you can find him in #gentoo-qt and #gentoo-kde.

**Comment by mndar** [comment]
Actually wanted to chat with you about the whole "adding qt5_add_dep() to qmake-utils.eclass" option.

If kde5-functions.eclass is not to be used then the ebuild needs two changes you recommended; dropping qt deps since everything in the portage tree is above that and removing that elog line.

Also qt5ct now has a new version, 0.23
I'm new to this, so how is a new version supposed to be handled?

**Comment by kensington** [comment]
IMO there's no value in adding a ``qt5_add_dep`` to ``qmake-utils.eclass``. It was only ever added to ``kde5-functions.eclass`` to make it easier to avoid peculiarities in portage's autounmask feature without repeated modification to hundreds of similar packages.

I agree with @Pesa's original advice.

**Comment by kensington** [comment]
To handle the new version, I suggest squashing your changes into one commit and then force-pushing to your PR branch.

File: x11-misc/qt5ct/qt5ct-0.22.ebuild [hunk 4]
```diff
@@ -12,9 +12,6 @@ SRC_URI="mirror://sourceforge/${PN}/${P}.tar.bz2"
 LICENSE="BSD"
 SLOT="0"
 KEYWORDS="~amd64"
-RDEPEND=">=dev-qt/qtcore-5.4.0
-	>=dev-qt/qtgui-5.4.0
-	>=dev-qt/qtwidgets-5.4.0"
```
Comment (Pesa) [inline x11-misc/qt5ct/qt5ct-0.22.ebuild]: what?

**Comment by Pesa** [comment]
You did not squash the commits...

**Comment by mndar** [comment]
I'm closing this PR because there are a lot of mistakes done here. Will redo the whole process.

**Pull request closed without merging**